File: 2020/Day-01/main.dart
import "dart:convert";
import "dart:io";

void part1(List<int> vals) {
  outer:
  for (var i in vals.sublist(0, vals.length - 1)) {
    for (var j in vals.sublist(1)) {
      if (i + j == 2020) {
        print(i * j);
        break outer;
      }
    }
  }
}

void part2(List<int> vals) {
  outer:
  for (var i in vals.sublist(0, vals.length - 2)) {
    for (var j in vals.sublist(1, vals.length - 1)) {
      for (var k in vals.sublist(2)) {
        if (i + j + k == 2020) {
          print(i * j * k);
          break outer;
        }
      }
    }
  }
}

Future<void> main(List<String> args) async {
  var vals = await File("../input01.txt")
      .openRead()
      .map(utf8.decode)
      .transform(LineSplitter())
      .map(int.parse)
      .toList();
  part1(vals);
  part2(vals);
}

File: 2020/Day-01/main.pl
#!/usr/bin/perl
use strict qw(vars subs);
use warnings;
use v5.16;

sub part1 {
	my (@vals) = @_;
	outer:
	for my $i (@vals[0..$#vals - 1]) {
		for my $j (@vals[1..$#vals]) {
			if ($i + $j == 2020) {
				say $i * $j;
				last outer;
			}
		}
	}
}

sub part2 {
	my (@vals) = @_;
	outer:
	for my $i (@vals[0..$#vals - 2]) {
		for my $j (@vals[1..$#vals - 1]) {
			for my $k (@vals[2..$#vals]) {
				if ($i + $j + $k == 2020) {
					say $i * $j * $k;
					last outer;
				}
			}
		}
	}
}

my @vals = map s/\s+$//r, do {
	local @ARGV = ("../input01.txt");
	<>;
};
part1(@vals);
part2(@vals);
